packages/session-recorder/src/index.ts

```diff
@@ -206,6 +204,13 @@ const SplunkRumRecorder = {
 						return
 					}
 
+					// We need to call it here before another getSessionID call. This will create a new session
+					// if the previous one was expired
+					if (SplunkRum._internalOnExternalSpanCreated) {
```

Comment:
The hook is called only when we are sure the session recorder chunk is not discarded.